### PR TITLE
Pthreads: Fix confusion between main browser and main runtime thread

### LINKED ISSUE #11914: pthread_create doesnt fire when ENVIRONMENT_IS_WORKER
Hello,
Been trying to create a pthread pool with emscripten, but having issues with getting the threads started when calling it from a web worker.

A small example `pthread_test.c`:
```C
#include <pthread.h>
#include <stdio.h>
void* tTask(void* arg) {
  printf("hello \n");
  return NULL;
}

int run() {
  printf("Starting run\n");
  pthread_t pid;
  pthread_create(&pid, NULL, tTask, NULL);
  pthread_join(pid, NULL);
  printf("Done\n");
  return 0;
}
```
with a simple html file that creates the worker `worker.js`
```JavaScript
importScripts('Wasmpool.js');
const Module = {
  locateFile: file => file,
  onRuntimeInitialized: () => Module._run(),
  mainScriptUrlOrBlob: 'Wasmpool.js',
};
WasmPool(Module);
```

tTask is called when exported using html as output, renaming run to main and adding the flag `-s PROXY_TO_PTHREAD`, but not in a worker as above. 
I tried it with the `latest` & `tot` builds and `Chrome Version 84.0.4147.125 64bits` + `Chrome Version 86.0.4234.0 (Official Build) canary`.
Any ideas?

Edit: Looks like it gets stuck [here](https://github.com/emscripten-core/emscripten/blob/master/src/worker.js#L203) , but the signature seems correct

### PR DESCRIPTION
**Pull request by kripken**
The main browser thread is the page with the DOM and HTML, while the
main runtime thread is where we start up the JS, and where things like
stdio proxying go (so all file access is on the same thread). Normally that
is the same place. However, a pthreads application started in a worker has
no access to the main browser thread. In that case `emscripten_is_main_browser_thread`
always returns false, while `emscripten_is_main_runtime_thread` returns
true (on the main thread).

This PR fixes some confusion there, where two locations checked the
main browser thread when they meant to check the main runtime thread.
In both cases they care about handling proxied calls. Without this fix,
the testcase added here just hangs, as we never handle proxied calls
to the main runtime thread (since we think we are never that thread).

Fixes #11914 (testcase in #12080)

### CODE REVIEW

File: system/lib/libc/musl/src/thread/__timedwait.c [hunk 1]
```diff
@@ -41,7 +41,7 @@ int __timedwait_cp(volatile int *addr, int val,
 
 #ifdef __EMSCRIPTEN__
 	double msecsToSleep = top ? (top->tv_sec * 1000 + top->tv_nsec / 1000000.0) : INFINITY;
-	int is_main_thread = emscripten_is_main_browser_thread();
+	int is_main_thread = emscripten_is_main_runtime_thread();
```
Comment (dschuff) [inline system/lib/libc/musl/src/thread/__timedwait.c:44]: Is the behavior of this function actually supposed to be different on the non-browser thread, where we can't actually sleep? Or is that difference handled somewhere else?

**Review by dschuff — COMMENTED** [review]
Thanks, I'm glad to see this improving. (I wouldn't be surprised if there are other places where this distinction isn't quite right, too).

I was wondering before, whether there is a use case for starting the runtime on a worker, but still having one of the pthreads be on the browser main thread. Although I can't really think of what that might be. I guess the user's pre-js/post-js goes on whatever JS context the module is initially instantiated from?

**Comment by dschuff** [comment]
I guess the issue in #11914 doesn't happen more often because it's more common for people to use PROXY_TO_PTHREAD which does keep the runtime on the main browser thread, and just runs main on a background thread.

File: system/lib/libc/musl/src/thread/__timedwait.c (hunk 1 above)
Comment (kripken) [reply system/lib/libc/musl/src/thread/__timedwait.c]: Oh, I think you're right, this change is incorrect. The logic here is actually checking whether we can do a synchronous wait or not, so it does care about the main browser thread.

The problem fixed in this PR is for proxying, not for sleeping. So this part is wrong and should not be needed. Verified the test passes without it locally.

**Comment by kripken** [comment]
> I was wondering before, whether there is a use case for starting the runtime on a worker, but still having one of the pthreads be on the browser main thread. Although I can't really think of what that might be.

Yes, I don't think we have a use case for that. PROXY_TO_PTHREAD is pretty close (there is main browser thread access, but isn't a pthread).

> I guess the user's pre-js/post-js goes on whatever JS context the module is initially instantiated from?

pre/post-js is added to the main JS, which is run on each JS environment, including on each pthread.

> I guess the issue in #11914 doesn't happen more often because it's more common for people to use PROXY_TO_PTHREAD which does keep the runtime on the main browser thread, and just runs main on a background thread.

Could be, yeah, and also that it's rare to run a pthreads application starting from a worker. The common case, I assume, is to run the application from the main HTML (with or without `PROXY_TO_PTHREAD`).

**Review by dschuff — APPROVED** [review]

**Pull request merged**